src/topology/separation.lean

```diff
@@ -554,3 +554,66 @@ begin
 end
 
 end normality
+
+/-- In a compact t2 space, the connected component of a point equals the intersection of all
+its clopen neighbourhoods. -/
+lemma connected_component_eq_Inter_clopen [t2_space α] [compact_space α] {x : α} :
+  connected_component x = ⋂ Z : {Z : set α // is_clopen Z ∧ x ∈ Z}, Z :=
+begin
+  apply eq_of_subset_of_subset,
+  { exact connected_component_subset_Inter_clopen},
+  { have hs : @is_closed _ _inst_1 (⋂ (Z : {Z : set α // is_clopen Z ∧ x ∈ Z}), ↑Z),
+    { exact is_closed_Inter (λ Z, Z.2.1.2) },
+    apply subset_connected_component,
+    { apply (is_preconnected_iff_subset_of_fully_disjoint_closed hs).2,
+      intros a b ha hb hab ab_empty,
+      haveI := @normal_of_compact_t2 α _ _ _,
+      rcases normal_separation a b ha hb (disjoint_iff.2 ab_empty) with ⟨u, v, hu, hv, hau, hbv, huv⟩,
+      suffices : ∃ (Z : set α), is_clopen Z ∧ x ∈ Z ∧ Z ⊆ u ∪ v,
+      { cases this with Z H,
+        rw [disjoint_iff_inter_eq_empty] at huv,
+        have H1 := is_clopen_inter_of_disjoint_cover_clopen H.1 H.2.2 hu hv huv,
+        rw [union_comm] at H,
+        rw [inter_comm] at huv,
+        have H2 := is_clopen_inter_of_disjoint_cover_clopen H.1 H.2.2 hv hu huv,
+        by_cases (x ∈ u),
+        { left,
+          suffices : (⋂ (Z : {Z : set α // is_clopen Z ∧ x ∈ Z}), ↑Z) ⊆ u,
+          { rw [inter_comm, ←set.disjoint_iff_inter_eq_empty] at huv,
+            replace hab : (⋂ (Z : {Z // is_clopen Z ∧ x ∈ Z}), ↑Z) ≤ a ∪ b := hab,
+            replace this : (⋂ (Z : {Z // is_clopen Z ∧ x ∈ Z}), ↑Z) ≤ u := this,
+            exact disjoint.left_le_of_le_sup_right hab (huv.mono this hbv) },
+          { apply subset.trans _ (inter_subset_right Z u),
+            apply Inter_subset (λ Z : {Z : set α // is_clopen Z ∧ x ∈ Z}, ↑Z)
+            ⟨Z ∩ u, by {split, exact H1, apply mem_inter H.2.1 h}⟩ } },
+        have h1 : x ∈ v,
+        { cases (mem_union x u v).1 (mem_of_subset_of_mem (subset.trans hab
+            (union_subset_union hau hbv)) (mem_Inter.2 (λ i, i.2.2))) with h1 h1,
+          { exfalso, apply h, exact h1},
+          { exact h1} },
+        right,
+        suffices : (⋂ (Z : {Z : set α // is_clopen Z ∧ x ∈ Z}), ↑Z) ⊆ v,
+        { rw [←set.disjoint_iff_inter_eq_empty] at huv,
+          replace hab : (⋂ (Z : {Z // is_clopen Z ∧ x ∈ Z}), ↑Z) ≤ a ∪ b := hab,
+          replace this : (⋂ (Z : {Z // is_clopen Z ∧ x ∈ Z}), ↑Z) ≤ v := this,
+          exact disjoint.left_le_of_le_sup_left hab (huv.mono this hau) },
+        { apply subset.trans _ (inter_subset_right Z v),
+          apply Inter_subset (λ Z : {Z : set α // is_clopen Z ∧ x ∈ Z}, ↑Z)
+          ⟨Z ∩ v, by {split, exact H2, apply mem_inter H.2.1 h1}⟩ } },
+      have H1 := (is_compact.inter_Inter_nonempty (is_closed.compact (is_closed_compl_iff.2 (is_open_union hu hv)))
+          (λ Z : {Z : set α // is_clopen Z ∧ x ∈ Z}, Z) _),
```

Comment:
After this `have`, you have two goals, and I can't tell where you are proving the main goal and where you are proving `H1`. Why don't you put the proof of `H1` just after its statement?

Also, in a longish proof like that, adding comments explaining where you are in the proof improves readability a lot.

Also, isn't the `H1` line too long?

Reply:
Fixed! I actually forgot that H1 produced a second goal, it is proven in the final exact of the lemma. I replaced the underscore of H1 by the final exact, so that H1 doesnt produce two goals anymore.